src/GraphQL/Validation/IValidationResult.cs

```diff
@@ -14,5 +18,26 @@ public interface IValidationResult
         /// Returns a list of the errors found during validation of a document.
         /// </summary>
         ExecutionErrors Errors { get; }
+
+        /// <summary>
+        /// Returns the set of variables parsed from the inputs.
+        /// If the document did not pass validation, this value will be <see langword="null"/>.
+        /// If the document passed validation but contained no variables, this value will be <see cref="Variables.None"/>.
+        /// </summary>
+        Variables? Variables { get; }
+
+        /// <summary>
+        /// Returns a dictionary of fields, and for each field, a dictionary of arguments defined for the field with their values.
+        /// If the document did not pass validation, or if no field arguments were found, this value will be <see langword="null"/>.
+        /// Note that fields will not be present in this dictionary if they would only contain arguments with default values.
```

Comment:
```
 // if any arguments were supplied in the document for the field, load all defined arguments
            // for the field
            if (field.Arguments?.Count > 0)
```

Is this controlled by this piece of code? And the purpose is to return `FieldDefinition.DefaultArguments` here

> _executionContext.ArgumentValues?.TryGetValue(FieldAst, out var ret) ?? false ? ret : FieldDefinition.DefaultArguments;

without allocating additional dictionary. Right?

Reply:
I believe that's correct, preventing allocation of the dictionary and their entries as well.  So basically, we are optimizing for the idea that much of the time there are no field arguments specified, and simply the default set needs to be returned.  If I remember correctly, this is a performance improvement over the existing codebase.